### DISCUSSION TITLE
Is there a way to keep models in memory?

### DISCUSSION THREAD

**Discussion by KaruroChori** [Comfy-Org/ComfyUI#311] [Q&A]
It seems to me that for each task models are loaded and offloaded from the GPU memory at each task execution.
Would it be possible to keep them cached and make the generation process faster as long as they are still the same?
Because at the moment this seems to lead to a massive performance hit compared to auto1111, at least on my system.

**Comment by comfyanonymous** [comment]
You can use this command line option to always keep them in memory: --highvram

**Reply by KaruroChori** [reply]
Thank you. I am still not getting the same degree of performance, but this issue is resolved.
I am testing comfyui on a tesla P40, would you expect any specific problem with this generation of cards?

**Reply by comfyanonymous** [reply]
Can you give me more details of how you are running both? Are you using the CheckpointLoaderSimple node in ComfyUI? Are you using the standalone build or did you install manually?

**Reply by KaruroChori** [reply]
I would love to report some quantitative data, but I made the mistake to update Automatic1111 earlier today, and it is now broken. So I cannot provide proper data :/

Yes, I am using CheckpointLoaderSimple.
It is currently installed via docker-compose, as one of the services in https://github.com/jkovacs-dev/stable-diffusion-webui-docker/tree/comfyui.
When I say slower I do not mean an order of magnitude worse, probably around 2 or 3 times less.

For reference a tesla P40 is only natively supporting fp32 and int8.
fp16 is only simulated from what I understand, and as such slower.

**Reply by tiago-falves** [reply]
I have a problem in which I have two workflows, which I run in turns.
The problem is that if I run the first workflow, it loads the model. I then run the second workflow, and when the first workflow again, it loads the model again. Is there a way to keep the model loaded? 
With the --highvram option, it loads the model again.
Thanks in advance! @comfyanonymous

**Reply by Expertaz** [reply]
@tiago-falves Hey ! I have two workflows, which I run in turns too. Have you found any solution ? Thanks :)

**Reply by tiago-falves** [reply]
Hey!
The solution is unifying both workflows, and change the first connection, depending on which workflow you want to use! 
This way it keeps both models loaded!

**Reply by Expertaz** [reply]
@tiago-falves haha thanks! Could you please spare one more minute to explain how to change the initial connection depending on the workflow I intend to use? That would be appreciated (:

**Reply by tiago-falves** [reply]
1. Place Both Workflows on the Same Page: Begin by arranging both workflows on the same workspace in Comfy.
2. Shared "Base Model" Node: Both workflows will use the same "Base Model" node. This node acts as a starting point for the data processing or analysis performed in each workflow. This is just an example, the same logic would apply if you would have different models, but one "preview image" node in common for example.
3. Connecting the Base Model to a Workflow: To run a workflow, connect the output of the "Base Model" node to the starting point (beginning node) of the desired workflow. This connection directs the output data from the model into the selected workflow for further processing.
4. Running the Workflow: Once the connection is made, execute the workflow. The data will flow from the "Base Model" through the connected workflow, processing as specified within that workflow's sequence of nodes.
5. Switching Workflows: If you need to switch to the other workflow:
	a. Disconnect the Current Workflow: First, disconnect the "Base Model" node from the currently active workflow. This stops data flow to this workflow.
	b. Connect to the New Workflow: Next, connect the "Base Model" node’s output to the beginning of the other workflow. This initiates data flow into the new workflow.

**Reply by UVMFilm** [reply]
Ever since you're new update it doesn't do this anymore so now it takes double the amount of time to process videos ... any work around? Why vram isn't doing it it's still offloading after every prompt

**Comment by efwfe** [comment]
Hello, try this. https://github.com/comfyanonymous/ComfyUI/pull/3545

**Comment by ghost** [comment]
How can you achieve this when using the API?

**Reply by AI-taja** [reply]
Did you find a solution?

**Reply by ghost** [reply]
> Did you find a solution?

Actually, yes, check out Inspire Shared Checkpoint loader. These nodes are amazing, exactly what we needed.

With his nodes, you can actually cache anything, loras, checkpoints, models, latents, anything! One of my favorite node packs.
https://github.com/ltdrdata/ComfyUI-Inspire-Pack
